src/components/media-views.js

```diff
@@ -837,3 +846,150 @@ AFRAME.registerComponent("media-image", {
     this.el.emit("image-loaded", { src: this.data.src, projection: projection });
   }
 });
+
+AFRAME.registerComponent("media-pdf", {
+  schema: {
+    src: { type: "string" },
+    projection: { type: "string", default: "flat" },
+    contentType: { type: "string" },
+    index: { default: 0 },
+    batch: { default: false }
+  },
+
+  init() {},
```

Comment:
No need to include empty methods, I believe aframe checks for them before attempting to call them.